build-common/src/cbindgen.rs

```diff
@@ -67,8 +59,26 @@ pub fn generate_and_configure_header(header_name: &str) {
         deliverables_dir = parent_dir.join(&deliverables_dir);
     }
 
+    (cargo_target_dir, deliverables_dir)
+}
+
+/// Configure the header generation using environment variables.
+/// call into `generate_header` with the appropriate arguments.
+///
+/// Expects CARGO_MANIFEST_DIR to be set.
+/// If DESTDIR is set, it will be used as the base directory for the header file.
+/// DESTDIR can be either relative or absolute.
+/// Either CARGO_TARGET_DIR is set, or `cargo locate-project --workspace` is used to find the base
+/// of the target directory.
+///
+/// # Arguments
+///
+/// * `header_name` - The name of the header file to generate.
+pub fn generate_and_configure_header(header_name: &str) {
```

Comment:
Do we want this to panic on failure, or return an `Error`

Reply:
In a build script, I feel the panic option is reasonable, though I will defer to you on such questions.